client_end_user/shared/src/commonMain/kotlin/presentation/search/SearchScreen.kt

```diff
@@ -1,15 +1,183 @@
+        when (effect) {
+            is SearchUiEffect.NavigateToMeal -> TODO()
+            is SearchUiEffect.NavigateToRestaurant -> navigator.root?.push(RestaurantScreen)
+        }
+    }
+
+    @OptIn(ExperimentalResourceApi::class, ExperimentalFoundationApi::class)
+    @Composable
+    override fun onRender(state: SearchUiState, listener: SearchInteractionListener) {
+        LazyColumn(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(Theme.colors.background)
+                .padding(getStatusBarPadding())
+                .padding(top = 16.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            contentPadding = PaddingValues(bottom = 16.dp)
+        ) {
+
+            stickyHeader {
+                BpSimpleTextField(
+                    state.query,
+                    hint = Resources.strings.searchHint,
+                    hintColor = Theme.colors.contentSecondary,
+                    onValueChange = listener::onSearchInputChanged,
+                    onClick = {},
+                    leadingPainter = painterResource(Resources.images.searchOutlined),
+                    modifier = Modifier.background(Theme.colors.background).padding(horizontal = 16.dp)
+                )
+            }
+
+            item {
+                LazyRow(
+                    contentPadding = PaddingValues(horizontal = 16.dp),
+                    horizontalArrangement = Arrangement.spacedBy(16.dp)
+                ) {
+                    items(state.restaurants) {
+                        Restaurant(
+                            restaurant = it,
+                            onClick = listener::onRestaurantClicked
+                        )
+                    }
+                }
+            }
+
+            items(state.meals) {
+                MealCard(
+                    meal = it,
+                    onClick = listener::onMealClicked
+                )
+            }
+        }
+    }
+
+
+    @Composable
+    private fun Restaurant(
+        modifier: Modifier = Modifier,
+        restaurant: RestaurantUiState,
+        onClick: (String) -> Unit
+    ) {
+        Column(
+            modifier = modifier.widthIn(max = 64.dp).noRippleEffect { onClick(restaurant.id) },
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Image(
+                modifier = Modifier.roundedBorderShape().size(64.dp),
+                painter = rememberAsyncImagePainter(restaurant.image),
+                contentScale = ContentScale.Crop,
+                contentDescription = restaurant.name,
+            )
+
+            Text(
+                text = restaurant.name,
+                style = Theme.typography.body,
+                color = Theme.colors.contentPrimary,
+                maxLines = 2,
+                textAlign = TextAlign.Center
+            )
```

Comment:
![image](https://github.com/TheChance101/beep-beep/assets/83292287/b972b01c-c587-489d-84b3-1a690c477cbb)
**we just need an overflow ellipse**